TESTDATA_GUIDELINES.md

```diff
@@ -0,0 +1,277 @@
+data is probably illegal under the GDPR, or at the very least a possible problem.
+
+When you create your own test data with dummy accounts, you can rest assured there won't be any problems, as you're not
+using real personal data.
+
+⚠️ Be careful even when creating your own data! Some applications will automatically store data from other users (for
+example public chat channels or "users nearby" features). If these features are included in your apps, you may still
+need consider the GDPR.
+
+### Exotic applications
+
+Lastly, the most obvious point: some applications simply aren't that common, so finding data from them online or from
+other sources is not an option.
+
+## 1. Plan ahead
+
+Now that you're convinced to create your own test data, let's get to it. The first step is careful planning. You might
+be tempted to just start sending messages or start browsing in the app you want to generate reference data for, but from
+experience we learned creating test data is something best properly planned ahead.
+
+### Investigate the application
+
+As [stated before](#Completeness), apps these days have MANY features. Therefore, it's a good idea to explore the
+application you want to create test data with. Even if you use this app privately, you'll probably find feature you
+never knew.
+
+If you're creating the reference data for a specific goal (forensic research, network analysis, or something else),
+keep this goal in mind: if you're only doing forensic research on pictures, there's probably no need to delve into the
+details of calendar appointments... Although: make sure to check if those appointments can contain a picture or
+thumbnail!
+
+The goal of this investigation is having a concise list of application features you want to cover.
+
+### Create a script
+
+After you have a list of all features you want to cover, it's time to write a script. And this isn't necessarily a Puma
+script: Puma can help you to automate actions, but if your reference data is about an app that Puma doesn't support, and
+you're just going to cover 10 user actions, you'll probably save time by executing the actions manually to create your
+reference data.
+
+Still, even when you create the data manually, write down a **precise**, **chronological** and **complete** description
+of all actions you will take to create the reference data.
+
+By **precise** we mean there should be no room for ambiguity. `send a picture in whatsapp` is not precise: does this
+mean using the embedded camera to take a picture and send it, or does it mean sending a picture stored on the device?
+Be sure every action in your script unambiguously describes one specific user action in the application.
+
+By **complete** we mean you should obviously include all actions you will take, but also actions you do not take: after
+sending a picture from Alice to Bob, include in your script whether or not Bob opens the picture. When making a call
```

Comment:
This is the first mention of Alice and Bob, and it's not very clear where the names come from. Maybe explain it or use something like `user 1` and `user 2`?